cumulusci/tasks/sfdx.py

```diff
@@ -95,3 +96,17 @@ def _get_command(self):
 
     def _process_data(self, data):
         self.logger.info("JSON = {}".format(data))
+
+
+class ChooseDeploymentMethod(BaseTask):
```

Comment:
@davisagli sent me this PR, so, he asked for it... ;)

I'm _not_ in love with the idea of a task that just provides facts about the environment. I see the above comments from Jason, and I think youre right that building out a full blown expression syntax is going to take a little more effort. but, although clever (and I'm proud of CCIs ability to support this as implemented!)  I don't know that something that just calculates some boolean for a ret_val is a great use of a task. MetaCI and MetaDeploy are (or should be!) storing the task ret_vals and at some point highlighting relevant ones in the UI. This would be noisy.

Obviously ProjectConfig is... overloaded at the moment, does CCI need a modular way to provide project_config vars that are available to tasks?

However, as I said, pretty cool that you can do this with CCI tasks/ret_vals. don't not merge this on my architectural whinging.

Reply:
@cdcarter I agree with the critique, and I was also thinking along the lines of a modular way to define variables that are then available to tasks. (compare ansible's "facts": https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variables-discovered-from-systems-facts) That could even be used to construct a preflight plan for MetaDeploy based on which variables are referenced in conditional expressions for tasks.

But also I wanted to get this done without delaying it for big infrastructural work. Consider this a temporary pragamatic stopgap.